frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,479 @@
+        <tr>
+            <td className={checkForActiveColumn(ColumnDataEnum.Name)}>
+                {receiver.Name}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Service)}>
+                {receiver.Service}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Date)}>
+                {receiver.Date}
+            </td>
+            <td
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.ReportId
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.ReportId}:`,
+                        body: receiver.ReportId,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.ReportId}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Main)}>
+                <p
+                    className={`font-mono-sm border-1px bg-primary-lighter radius-md padding-left-1 padding-right-1 margin-top-0 ${
+                        uppercaseMain === StatusEnum.BATCH && "bg-blue-5"
+                    } ${uppercaseMain === StatusEnum.PROCESS && "bg-blue-10"} ${
+                        uppercaseMain === StatusEnum.READY && "bg-blue-20"
+                    }`}
+                >
+                    {uppercaseMain}
+                </p>
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Sub)}>
+                {receiver.Sub}
+            </td>
+            <td
+                className={`message-receiver-break-word white-space-pre ${checkForActiveColumn(
+                    ColumnDataEnum.FileName
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.FileName}:`,
+                        body: receiver.FileName,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.FileName}
+            </td>
+            <td
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.TransportResults
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.TransportResults}:`,
+                        body: receiver.TransportResults,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.TransportResults}
+            </td>
+        </tr>
+    );
 };
 
 export const MessageReceivers = ({ receiverDetails }: MessageReceiverProps) => {
+    const modalRef = useRef<ModalRef>(null);
+    const [modalText, setModalText] = useState({ title: "", body: "" });
+    const [activeColumn, setActiveColumn] = useState("");
```

Comment:
Do we need to save active column and active column sort order separately or could we just do something like storing them both in one state object?

```tsx
const [activeSortColumn, setActiveSortColumn] = useState(null);
// setting one column
setActiveSortColumn({ column: "Name", order: "ASC" });
// resetting
setActiveSortColumn(null);
```

I think it's easier to reason about if we treat all sort operations as an atomic change (both sorted column and sort order) 🤷

Reply:
I was thinking about that, but I've really only seen that pattern when updating a large state object with `useReducer`. From what I've seen in the app and in [React's docs](https://reactjs.org/docs/hooks-state.html#tip-using-multiple-state-variables), `useState` is super explicit in updating a very _specific_ bit of state even if it feels excessive.

Reply:
Etan and I chatted about this separately, and this isn't a huge deal either way.  My rationale was coming from state updates getting swallowed up [in React's batching](https://github.com/facebook/react/issues/14259) (like lots of `setState` calls sequentially), but it should be fine here since it's happening in an event-handler.